### DISCUSSION TITLE
Theia GIT-UI support

### DISCUSSION THREAD

**Discussion by arunswarnam** [eclipse-theia/theia#10753] [Q&A] [answered]
From what I understood, browser based Theia app doesn't support GIT UI interface to prompt for GIT username/password as noted here: https://github.com/eclipse-theia/theia/issues/3397

However I see there are more recent threads that suggest to use git-ui vscode extension as an alternate to using Theia extensions @theia/git and @theia/scm: https://community.theia-ide.org/t/how-to-integrate-git-in-webide

Is there a recommended way to use GIT in Theia browser based app that comes with UI support for prompting missing credentials if the underlying GIT is not configured? 

Also, Is there a reason why browser based Theia app (NativeTheia extensions) doesn't support GIT UI for prompting username and password as they do in electron based Theia?

**Comment by vince-fugnitto** [comment]
@arunswarnam thank you for the discussion :+1: 

If you were to use [vscode-builtin-git](https://open-vsx.org/extension/vscode/git/1.53.2) and [vscode-buiiltin-git-ui](https://open-vsx.org/extension/vscode/git-ui/1.53.2) does the feature not work? These are the same extensions that are present by default in vscode. These extensions replace `@theia/git` but `@theia/scm` needs to be kept in the application.

**Reply by arunswarnam** [reply]
@vince-fugnitto Thanks for the response. The features (prompt for username and password and few other UI polishes to other git operations)  work when using vscode-git and vscode-git-ui built in extensions but I also noticed that git history views are not showing up with the removal of @theia/git. I see this error "History is not supported for Git source control." 

1. Should we be adding vscode extensions like [this](https://marketplace.visualstudio.com/items?itemName=donjayamanne.githistory ) to get the desired git feature? 

2. As an aside, is there a recommendation from Theia team going forward? should users rely on vs code extensions for GIT features or should we be using @theia/* extensions for git features?

**Reply by vince-fugnitto** [reply]
@arunswarnam no problem!

1. You can definitely add extensions like [git history](https://open-vsx.org/extension/donjayamanne/githistory) and even ones like [gitlens](https://open-vsx.org/extension/eamodio/gitlens). Using the builtin git extensions (which come from vscode directly) enable such extensions to be used in the framework, which otherwise cannot with `@theia/git` (which was legacy git support before we were compatible with vscode extensions). The history you meantioned that does not work sounds like a bug and I'll open an issue for it however.

2. Going forward I would recommend the builtin git extensions over `@theia/git` since it allows the possibility to use other git-related vscode extensions.

**Reply by arunswarnam** [reply]
Thanks for clarifying. 

We are on v1.19.0 but I did try out the example app from v1.22.1.. It shows the same error:
<img width="1107" alt="image" src="https://user-images.githubusercontent.com/10571107/155369768-6d3349a1-01d3-4f2d-8079-6cd791c5c22f.png">
It would be great if you could share the issue details.

**Reply by msujew** [reply]
@arunswarnam Vince created an issue for this already here: https://github.com/eclipse-theia/theia/issues/10786. Support for scm-history extensions seems to be broken right now.

**Reply by arunswarnam** [reply]
Thank you @vince-fugnitto and @msujew

**Comment by vince-fugnitto** [comment, marked as answer]
@arunswarnam I remembered that if we are to use the vscode extensions for git support ([builtins](https://github.com/eclipse-theia/theia/discussions/10753#discussioncomment-2181297)) then we should not use `@theia/git` or `@theia/scm-extra` (extra scm funtionality not supported by the builtins such as history). For history we have `@theia/timeline` which provides the history or we can use git extensions mentioned at https://github.com/eclipse-theia/theia/discussions/10753#discussioncomment-2236115.

_`@theia/timeline`_:

![image](https://user-images.githubusercontent.com/40359487/155555143-675f263c-3bf7-4c72-9df5-383df9fe5e2b.png)
